### PR TITLE
Add models

### PR DESCRIPTION
**Pull request by QuangTranXuan**
Add models

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Category < ActiveRecord::Base
+  has_many :words, dependent: :destroy
+  has_many :lessons
+  has_many :users, through: :lessons
```
Comment (dieunb) [inline app/models/category.rb]: this relation `has_many :users, through: :lessons`

Comment (QuangTranXuan) [inline app/models/category.rb]: My code is exactly: has_many :users, through: :lessons!

Comment (QuangTranXuan) [inline app/models/category.rb]: Quan hệ này bị lỗi gì vậy anh?

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  has_many :lessons
+  has_many :categories, through: :lessons
+  validates :name, presence: true, length: {maximum: 50}
+
+  validates :password,  presence: true, length: { minimum: 6 }, unless: :not_validate_password
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(?:\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, format: { with: VALID_EMAIL_REGEX },
+            uniqueness: { case_sensitive: false }
+
+  before_save { self.email = email.downcase }
```
Comment (dieunb) [inline app/models/user.rb]: remove space before `}` and after `{`

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  has_many :lessons
+  has_many :categories, through: :lessons
+  validates :name, presence: true, length: {maximum: 50}
+
+  validates :password,  presence: true, length: { minimum: 6 }, unless: :not_validate_password
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(?:\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, format: { with: VALID_EMAIL_REGEX },
```
Comment (dieunb) [inline app/models/user.rb]: remove space before `}` and after `{`

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  has_many :lessons
+  has_many :categories, through: :lessons
+  validates :name, presence: true, length: {maximum: 50}
+
+  validates :password,  presence: true, length: { minimum: 6 }, unless: :not_validate_password
```
Comment (dieunb) [inline app/models/user.rb]: remove space before `}` and after `{`

**Pull request merged**